### DISCUSSION TITLE
Websocket+TLS+Web(Nginx)失败,之前一直可以的,不知道原因.

### DISCUSSION THREAD

**Discussion by HiColinn** [v2fly/v2ray-core#1113] [Q&A]
之前一直使用的这个组合,但是今天突然就不行了.不知道原因啊!!!希望大家帮看一下:
------

服务端环境:
  `OS:CentOS8`
   `v2Ray version:v4.40.1`
   `Nginx version:1.14.1`
客户端环境:
  `OS:Windows10`
  `v2ray GUI:v2rayN-V4.18(v2ray Core v4.40.1)`

------
附上配置:
v2ray配置:
```
// vmess-ws_tls_web入站
{
    "inbounds": [
        {
            "tag": "in_vmess_ws_tls_web",
            //只监听本地
            "listen": "127.0.0.1",
            "port": 7210,
            "protocol": "vmess",
            "settings": {
                "clients": [
                    {
                        "id": "c8f51c85-ea6d-6863-1e9c-fccd1b752036",
                        "alterId": 0,
                        "level": 0,
                        "email": "ZhangSan@gmail.com"
                    }
                ]
            },
            "streamSettings": {
                "network": "ws",
                "wsSettings": {
                    // 这里的path要和网关保持一致
                    "path": "/v2proxy"
                }
            }
        }
    ]
}
```

Nginx配置:

```
#转发v2ray代理
server {
  listen  443 ssl http2;
  server_name domain.com;
  ssl_certificate       /etc/nginx/ssl/domain.com_ecc/fullchain.cer;
  ssl_certificate_key   /etc/nginx/ssl/domain.com_ecc/domain.com.key;
  ssl_session_timeout 1d;
  ssl_session_cache shared:MozSSL:10m;
  ssl_session_tickets off;

  ssl_protocols         TLSv1.1 TLSv1.2 TLSv1.3;
  ssl_ciphers           ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
  ssl_prefer_server_ciphers off;


  location /v2proxy { # 与 V2Ray 配置中的 path 保持一致
    #if ($http_upgrade != "websocket") { # WebSocket协商失败时返回404
    #  return 404;
    #}
    proxy_redirect off;
    #假设WebSocket监听在环回地址的7210端口上
    proxy_pass http://127.0.0.1:7210;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
    proxy_set_header Host $http_host;
    # Show real IP in v2ray access.log
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  }
}
```
------
客户端配置:
![20210702210744](https://user-images.githubusercontent.com/65964332/124279102-a9217500-db79-11eb-99d8-465b15c342d7.jpg)
客户端日志:
```
[root@ tmp]# ./v2ray -c config.json 
V2Ray 4.41.0 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.16.5 linux/amd64)
A unified platform for anti-censorship.
2021/07/02 09:30:22 [Info] main/jsonem: Reading config: config.json
2021/07/02 09:30:23 [Warning] V2Ray 4.41.0 started
2021/07/02 09:31:07 127.0.0.1:58308 accepted //www.google.com:443 [proxy]
2021/07/02 09:32:29 [Warning] [1344642836] app/proxyman/outbound: failed to process outbound traffic > proxy/vmess/outbound: failed to find an available destination > common/retry: [transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://v2proxy.domain.com/v2proxy):  > dial tcp ip:443: i/o timeout] > common/retry: all retry attempts failed
```
------
日志情况:
当客户端连接到这个配置的时候,Nginx的access.log和error.log都没有任何日志反馈,感觉就像是客户端没有发起请求一样.v2ray的日志更是没有监听到任何请求

**Comment by horizon98** [comment]
大概率是443端口被封了，本地不挂代理测试一下端口就行。我也是这样，换了IP没几天又被封了，最近有不少人都是Vmess/Vless+WS+TLS配置被封了443端口。不清楚是哪里的问题，但是个人建议暂时不要使用这个配置了。

**Reply by HiColinn** [reply]
443没有被封,直接访问https://ip/path 是可以的,而且证书也加载正常,只不过返回的是404

**Reply by horizon98** [reply]
> 443没有被封,直接访问https://ip/path 是可以的,而且证书也加载正常,只不过返回的是404

我的情况是这样：使用V2ray发现无法连通节点后，关闭V2rayN，访问网站，无法连接。几分钟后，可以继续访问网站。如果继续使用V2ray，会好几秒钟，然后就无法连接了，关闭V2rayN之后一会儿端口又好了。端口被封期间，443端口可以ping通，但是使用tcping无法ping通。

我按这个配置用了大概2个月，没有更改过配置，就是从前几天开始的。

以上是我的情况，可能楼主的情况和我不一样吧。

**Reply by HiColinn** [reply]
卧槽!还真是,你不说我都不知道,刚才试了一下,
之所以我刚才说,可以访问,是因为我通过vmess+tcp的方式开启了正常代理,所以才能访问https开头的443端口的网址,当我关闭了v2rayN以后,就是直连访问我的nginx 443的时候,就无法访问了,果然是因为443被封.
另外我想说一下:
我大概两个月前从v2ray迁移到了xray,使用了xray的vless回落.让xray处理所有htpp的tls证书,然后正常流量再给到nginx,然后前几天就突然坏了,我尝试很久发现xray都无法使用(因为xray的回落要单独监听443,然后回给nginx),我以为是xray除了问题,今天又迁移回到了v2ray,还是采用用了很久的ws+tls+web的方式,发现也不行,怎么都没想到443端口呗封了
另外我发现xray项目已经好几个月不更新了,我怀疑是不是xray的vless回落被检测了?才导致443被封
你最近有没有用过xray,尤其是vless回落,即xray监听443

**Reply by horizon98** [reply]
那我也详细说一下我的情况吧：

没有用过Xray，我对V2ray其实也不是很熟悉。我的配置就是：v2ray的Vmess+ws+tls，使用Nginx部署了一个很简单的站点做伪装，用了Let's Encrypt的证书，监听80和443端口，其中80端口自动转发到443。部署时v2ray安装用的官方脚本，证书用certbot领取的，其他就没用过一键脚本了。Centos和Ubuntu上部署的都被封了。

然后客户端方面，基本就是电脑上的V2rayN、安卓上的V2rayNG以及路由器梅林固件的科学上网插件。之前两月一直都很正常，没有更换过任何配置，就前几天被封的。后来换了2次IP（非同一VPS），大概都是两天不到就检测到了。

**Comment by HiColinn** [comment]
破案了,原因是443被封,具体原因不明.

**Reply by ghost** [reply]
不仅443，换了其他端口后，过一阵子其他端口也会出现相同问题

**Comment by ghost** [comment]
我也这个问题
https://github.com/XTLS/Xray-core/discussions/634

**Reply by cnscorpions** [reply]
换成 1w+ 端口，好像也不稳 :<
